src/lib.rs

```diff
@@ -280,15 +282,18 @@ impl<TypeState> ConnectionOptions<TypeState> {
             stream: stream.try_clone()?,
             info: server_info,
             sid: AtomicUsize::new(1),
-            subs: Arc::new(RwLock::new(HashMap::new())),
-            pongs: Arc::new(Mutex::new(VecDeque::new())),
-            writer: Arc::new(Mutex::new(Outbound {
-                writer: BufWriter::with_capacity(64 * 1024, stream.try_clone()?),
-                flusher: None,
-                should_flush: true,
-                in_flush: false,
-                closed: false,
-            })),
+            shared_state: Arc::new(SharedState {
+                last_error: RwLock::new(Ok(())),
```

Comment:
This can be used for surfacing errors to users, and feeding the error information through the unified state object to anything else that may be doing IO.

Reply:
I like this, need one on subs with handlers too.